### PR TITLE
SR-6277 "WIP" Mute non-JSON output from "dump-package"

### PR DESCRIPTION
**Pull request by erwinmaza**
Submitting PR for feedback on this solution, in advance of writing unit tests.

This PR changes tool progress output to be muted by default.

It also introduces a new dump-package option: ` --include-tool-output` to override this default and show dependency resolution and fetch progress as before.

Diagnostics that contain an error are still printed to the console.

I would appreciate feedback on this solution as a new contributor to this codebase, not entirely familiar with its broader structure, and with expectations regarding tool output handling.

### CODE REVIEW

**Comment by hartbit** [comment]
@aciidb0mb3r Before @erwinmaza goes ahead with tests, I would have liked your input. During try! Swift, I suggested we mute the usual output (except on error diagnostic) instead of redirecting them to stderr because it felt weird to me to output non-error related output to stderr. What do you think?

File: Sources/Commands/SwiftPackageTool.swift [hunk 1]
```diff
@@ -175,6 +175,8 @@ public class SwiftPackageTool: SwiftTool<PackageToolOptions> {
             describe(graph.rootPackages[0].underlyingPackage, in: options.describeMode, on: stdoutStream)
 
         case .dumpPackage:
+            options.shouldMuteOutput = !options.dumpPackageOptions.includeToolOutput
+
             let graph = try loadPackageGraph()
             let manifest = graph.rootPackages[0].manifest
```
Comment (aciidb0mb3r) [inline Sources/Commands/SwiftPackageTool.swift]: Given that we're just dumping the root manifest, we shouldn't even use loadPackageGraph method to get the manifest. This will automatically avoid the problem of muting the resolution output (as there will be no resolution).

**Comment by erwinmaza** [comment]
@aciidb0mb3r @hartbit 

I took a look at avoiding calling `loadPackageGraph`. What I found was that code path does a lot of processing of pins, constraints, manifests, etc, so replicating that functionality to just get the dependency graph did not seem like a good approach. 

I then started down a path of passing down a `dryRun` parameter to eventually bypass the actual fetching of each dependency, when I discovered the `isResolverPrefetchingEnabled` property of SwiftTool, which is set by `options.shouldEnableResolverPrefetching`.  Setting that to `false` when running `dumpPackage` did prevent prefetching, but did not stop the actual fetch of each dependency.

Finally I found `DependencyResolver.merge()` calling `getContainer()`, which downloaded the dependency containers, which were then used to resolve subtrees: `let subtree = self.resolveSubtree(container, subjectTo: allConstraints, excluding: allExclusions)` 

Long story short: from what I can tell, fetching dependencies is necessary to obtain a compete dependency graph, and thus calling `loadPackageGraph` is unavoidable.

Please let me know if I'm missing something or if you had another approach in mind.

**Comment by aciidb0mb3r** [comment]
We basically need the manifest file of the root package. You can use the `loadRootManifests` method from `Workspace` to get the root package's manifest. Something like:

```
try getActiveWorkspace().loadRootManifests...
```

**Comment by aciidb0mb3r** [comment]
This looks good to me! I don't think we need any new tests here.

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Comment by hartbit** [comment]
Congrats @erwinmaza ! Sorry I took you on such a long-winded solution when there was a much simpler one 😔
